presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -849,4 +969,61 @@ public FunctionMap(FunctionMap map, Iterable<? extends SqlFunction> functions)
             return functions.get(name);
         }
     }
+
+    private class ApplicableFunction
+    {
+        private final Signature declaredSignature;
+        private final Signature boundSignature;
+
+        private ApplicableFunction(Signature declaredSignature, Signature boundSignature)
+        {
+            this.declaredSignature = declaredSignature;
+            this.boundSignature = boundSignature;
+        }
+
+        public Signature getDeclaredSignature()
+        {
+            return declaredSignature;
+        }
+
+        public Signature getBoundSignature()
+        {
+            return boundSignature;
+        }
+
+        /**
+         * One method is more specific than another if invocation handled by the first method could be passed on to the other one
+         */
+        public boolean isMoreSpecificThan(ApplicableFunction other)
```

Comment:
I tend to disagree, the fact that we don't use all properties of ApplicableFunction doesn't mean that the ordering of ApplicableFunctions isn't its property.
Of course this order makes sense only in context of FunctionRegistry, but so does the whole class and that is already represented by being inner class (doesn't matter whether static or not).